### PR TITLE
Update CI docu

### LINKED ISSUE #806: E2E: Mention test reports/results in docu
### Tasks

_No response_

### Use case

_No response_

### Related Pull request

_No response_

### LINKED ISSUE #844: improve ci.md
### Tasks

 - [ ] VITE_BASE_API_URL set /
 - [x] E2E in a list,
 - [x] E2E link/videos
 - [x] link checker
 - [x] sanity
 - [x] spell checker
 - [x] What is already my pre-commit?
 - [x] check again what is done in Jenkinsfile

### Use case

_No response_

### Related Pull request

_No response_

### PR DESCRIPTION
**Pull request by 4ydan**
closes #844
mentions about test results an reports in documentation, closes #806

## Basics

- [ ] I added a line to [/doc/CHANGELOG.md](/doc/CHANGELOG.md)
- [ ] The PR is rebased with current master.
- [ ] Details of what you changed are in commit messages.
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildserver is happy.

## Checklist

- [ ] I have installed and I am using [pre-commit hooks](../doc/contrib/README.md#Hooks)
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed the introduction tour
- [ ] I wrote migrations in a way that they are compatible with already present data
- [ ] I fixed all affected decisions
- [ ] I added unit tests for my code
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](https://github.com/ElektraInitiative/PermaplanT/blob/master/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
      (exceptions are documented)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] Examples are well chosen and understandable

### CODE REVIEW

File: doc/ci.md [hunk 1]
```diff
@@ -2,60 +2,66 @@
...
 
 Every pull request will be deployed on a publicly available instance on [pr.permaplant.net](https://pr.permaplant.net).
+Jenkins will acquires a lock here and releases it after finishing the E2E tests.
 
 Since there is only one agent for PRs available, the last built PR wins.
 
-### Deploy to Dev Environment
+### E2E Tests
+
+E2E tests are run on [dev.permaplant.net](https://dev.permaplant.net).
```
Comment (markus2330) [inline doc/ci.md:51]: Isn't it pr only?

```suggestion
E2E tests are only executed on [pr.permaplant.net](https://pr.permaplant.net).
```

File: doc/ci.md [hunk 2]
```diff
@@ -2,60 +2,66 @@
...
+If a test timeouts it is retried.
+
+Test reports and results are generated after successful execution.
+
+- A [cucumber report](https://build.libelektra.org/job/PermaplanT/job/master/lastCompletedBuild/cucumber-html-reports/overview-features.html)
+- A [html report](https://build.libelektra.org/blue/organizations/jenkins/PermaplanT/detail/master/395/artifacts)
```
Comment (markus2330) [inline doc/ci.md:57]: Why 395? Maybe also point to last completed build?

File: doc/ci.md [hunk 3]
```diff
@@ -2,60 +2,66 @@
...
 
 Since there is only one agent for PRs available, the last built PR wins.
 
-### Deploy to Dev Environment
+### E2E Tests
+
+E2E tests are run on [dev.permaplant.net](https://dev.permaplant.net).
+If a test timeouts it is retried.
+
+Test reports and results are generated after successful execution.
```
Comment (markus2330) [inline doc/ci.md:54]: Please describe how to find them when coming from "Details" of GitHub, i.e., e.g., for 846 https://build.libelektra.org/blue/organizations/jenkins/PermaplanT/detail/PR-846/5/pipeline

File: doc/ci.md [hunk 4]
```diff
@@ -2,60 +2,66 @@
 
 ## Triggers
 
-The pipeline will be executed for every Pull Request and every push to the repository.
+The pipeline will be executed ONLY for pushes to Pull Requests.
```
Comment (markus2330) [inline doc/ci.md:5]:
```suggestion
The pipeline will be executed **only** for pushes to Pull Requests and for pushes to master.
```

File: doc/ci.md [hunk 5]
```diff
@@ -2,60 +2,66 @@
...
 
-If you want the pipeline to be executed again on a Pull Request, you can add a comment saying `jenkins build please`.
-This will most likely be the case if you suspect that something is wrong with the build server infrastructure.
+If you want the pipeline to be executed, you can add a comment saying `jenkins build please`.
```
Comment (markus2330) [inline doc/ci.md:7]: Better describe when/where this should be done. (In PRs, if there are spurious problems)

File: doc/ci.md [hunk 6]
```diff
@@ -2,60 +2,66 @@
...
-- Test and Build Backend (With PostGIS sidecar container)
-- Test and Build Frontend
+- Parallel cargo stages
+- Sequential frontend stages
+- Sequential mdbook stages
 
-For more information about automated integration tests look [here](./tests/README.md).
+Following `hidden` checks are done:
 
-Every pipeline run has its own isolated database.
```
Comment (markus2330) [inline doc/ci.md:40]: Is this not true anymore?

File: doc/ci.md [hunk 7]
```diff
@@ -2,60 +2,66 @@
...
 
-- Test and Build Mdbook
-- Test and Build Backend (With PostGIS sidecar container)
-- Test and Build Frontend
+- Parallel cargo stages
+- Sequential frontend stages
+- Sequential mdbook stages
 
-For more information about automated integration tests look [here](./tests/README.md).
+Following `hidden` checks are done:
```
Comment (markus2330) [inline doc/ci.md:37]: What do you mean with `hidden`?

File: doc/ci.md [hunk 8]
```diff
@@ -2,60 +2,66 @@
...
-
-- prepare an empty PostgreSQL database with the PostGIS extension installed
-- call `./ci/build-scripts/build-backend.sh`, which builds the rust binary as well as the bindings' definition via `typeshare`
-- call `./ci/build-scripts/build-frontend.sh`, which does a full release build of the frontend
-
-### Deploy to PR Environment
+### Deploy PR
 
 Every pull request will be deployed on a publicly available instance on [pr.permaplant.net](https://pr.permaplant.net).
+Jenkins will acquires a lock here and releases it after finishing the E2E tests.
```
Comment (markus2330) [inline doc/ci.md:45]:
```suggestion
Jenkins will acquire a lock during execution of E2E tests.
```

File: doc/ci.md [hunk 9]
```diff
@@ -2,60 +2,66 @@
...
+- A [cucumber report](https://build.libelektra.org/job/PermaplanT/job/master/lastCompletedBuild/cucumber-html-reports/overview-features.html)
+- A [html report](https://build.libelektra.org/blue/organizations/jenkins/PermaplanT/detail/master/395/artifacts)
+- Videos and screenshots can on failed tests can also be found where the html report is
+
+### Deploy Dev
 
 The `master` branch will be automatically deployed to [dev.permaplant.net](https://dev.permaplant.net).
 
-### Deploy to Prod Environment
+### Deploy Prod
```
Comment (markus2330) [inline doc/ci.md:64]: Actually, here should be only written that the release pipeline is only relevant for doc/release.md (if this section is needed at all)

**Review by markus2330 — CHANGES_REQUESTED** [review]
Thank you, good job. A few things are still unclear to me, please improve. Please also request reviews of someone else for such documentation improvements. They should check if the text is understandable for someone new. (@horenso @tschawax)

File: doc/ci.md (hunk 1 above)
Comment (4ydan) [reply doc/ci.md:51]: Yes they get performed only on pr.permaplant.net.
Maybe we would want to run them on dev.permaplant.net when merged to master?

File: doc/ci.md (hunk 7 above)
Comment (4ydan) [reply doc/ci.md:37]: Some tests that you would not guess that they are there. This is actually an issue and should be made more transparent.

File: doc/ci.md (hunk 6 above)
Comment (4ydan) [reply doc/ci.md:40]: It is true, but I removed some information that I think is obvious or not important for the reader.

**Review by markus2330 — APPROVED** [review]
Looks much better, a few final suggestions.

File: doc/ci.md (hunk 1 above)
Comment (4ydan) [reply doc/ci.md:56]: Oh I just realized thats a type, I meant pr.permaplant

File: doc/ci.md [hunk 10]
```diff
@@ -2,64 +2,73 @@
...
 Since there is only one agent for PRs available, the last built PR wins.
 
-### Deploy to Dev Environment
+### E2E Tests
+
+E2E tests are run on [dev.permaplant.net](https://dev.permaplant.net).
+If a test times out, it is retried.
+
+[Test reports and results](https://build.libelektra.org/job/PermaplanT/job/master/lastCompletedBuild/artifact/e2e/) are generated on successful execution.
+They contain videos, screenshots and a html report.
```
[the remark this answers is not in the document]
Comment (4ydan) [reply doc/ci.md]: That test just timed out and didnt even connect to the webapp.

File: doc/ci.md (hunk 3 above)
Comment (4ydan) [reply doc/ci.md]: Is it well enough and clearly described now?

**Comment by markus2330** [comment]
@horenso @tschawax @temmey can you please check if everything is clear?

File: doc/ci.md [hunk 11]
```diff
@@ -2,64 +2,75 @@
...
-For users with login credentials for Jenkins, you can manually execute the pipeline for a branch or pull request via the [Jenkins UI](https://build.libelektra.org).
-Please ask if you want to have login data.
+For users with login credentials for Jenkins, you can manually execute the pipeline for a branch or pull request via the [Jenkins UI](https://build.libelektra.org/job/PermaplanT/).
+Please ask if you need login data.
 
-## Cancel Concurrent Builds
+## Concurrent Builds
 
-Previous builds on the same branch will get aborted to save computing time on the nodes.
+Concurrent builds on the same branch are disabled.
```
[the remark this answers is not in the document]
Comment (4ydan) [reply doc/ci.md:15]: Only the newest runs.
Yes manual if needed and you have the permissions.

File: doc/ci.md [hunk 12]
```diff
@@ -2,64 +2,75 @@
 
 ## Triggers
 
-The pipeline will be executed for every Pull Request and every push to the repository.
+The pipeline will be executed **only** for pushes to Pull Requests and for pushes to master.
```
[the remark this answers is not in the document]
Comment (markus2330) [reply doc/ci.md:5]: I think he would like to have a clarification that **all** pushes to Pull Requests get built, maybe something like:

```suggestion
The pipeline will be executed for:
- all pushes to branches for which a Pull Requests exists, starting with the creation of the pull request.
- all pushes to master.
```

@tschawax does my text clarify what you wanted to know?

File: doc/ci.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (markus2330) [reply doc/ci.md:15]: I answered in #874.

**Pull request merged**